src/overlay/Peer.cpp

```diff
@@ -137,6 +141,39 @@ Peer::MsgCapacityTracker::getPeer()
     return mWeakPeer;
 }
 
+bool
+Peer::isPeerSlow()
+{
+    auto now = mApp.getClock().now();
+    if ((now - mLastLatencyCheck) > LATENCY_CHECK_FREQUENCY)
+    {
+        mLatencyReasonable =
+            mPeerMetrics.mConnectionLatency.GetSnapshot().get95thPercentile() <=
```

Comment:
peer latency used like this is going to be pretty bad.
I pulled this from the last few hours:
![image](https://user-images.githubusercontent.com/4213763/202066834-0e2047e4-1001-4164-b2e1-304e9d1f640c.png)

Here are a few things to take away from this:
* latency can get very high - like minutes. This may cause to classify as "good" most peers (I think those peers end up getting dropped by the straggler logic)
* by the time we adjust latency for a peer, if it was "fast" before, it will have contributed a lot of bad data to the "fast" partition
* we should probably take into account `mPingSentTime` so that we can "fast fail" peers that don't respond.
* because you're using medida's window, you will often ignore high latency peers (in any 30 seconds window, you will not observe slow peers every time)
* (maybe related to the previous point) we probably want the classification to be relatively stable over time, ideally a peer that gets stuck for 20 seconds every 5 minutes but is fast the rest of the time should just be "bad" quality all the time (if other peers do not do that)
* maybe "ping latency" is too broken on its own. There are a bunch of other ways that peers can "go bad" (see `recurrentTimerExpired`), so it might be more reliable to aggregate most of those things into a norm of sorts, so that every "tick" we can "credit" badness to peers.